SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/polygon_cube_intersection/fast_polygon_cube_intersection.cpp

```diff
@@ -70,7 +70,7 @@ namespace polygon_cube_intersection
 
 #define TEST_AGAINST_PARALLEL_PLANES(posbit, negbit, value, limit)	\
 	if (mask & (posbit|negbit)) {					\
-        /*register*/ real temp = value;				\
+        /*register*/ float temp = value;				\
```

Comment:
Here the function is not templated. So real denotes a type, which was previously defined as a float. The components using those functions assume the parameters are of type float (check SparseGridTopology.cpp, line 817). That is why I chose float to replace real.
SReal is not an option neither because it can be switched between float and double. By default it is a double, so it makes incompatible the use of those functions with float.
A third option would be to template those functions. But we would also need to change all the components using them (e.g. SparseGridTopology).

If we look at SparseGridTopology, I'd say those function must take at least double as parameters (float also if templated). But it would require to modify some code calling those functions.